oeps/oep-0038-Data-Modeling.rst

```diff
@@ -0,0 +1,411 @@
+information. The purpose of Business Intelligence is to support better
+business decision making.
+
+**Composite Key** - In the context of relational databases, a composite key
+is a combination of two or more columns in a table that can be used to
+uniquely identify each row in the table. Uniqueness is only guaranteed
+when the columns are combined; when taken individually, the columns do
+not guarantee uniqueness.
+
+**CUD** - Create, Update, Delete. These are some of the actions that may
+happen to data in a database system.
+
+**CRUD** - Create, Read, Update, Delete. Same as CUD but with the *Read* operation added.
+
+**Data Best Practices** - These practices are designed to help teams create
+rich and efficient data models within the Open edX ecosystem. They are not
+standards but guidelines to help teams think about how to store data.
+
+**Data Dimension** - A Data Dimension is a set of data attributes pertaining
+to something of interest to a business. Dimensions are things like
+"customers", "products", "stores" and "time". For users of Data
+Warehouses, data dimensions are entry points to numeric facts (e.g.
+sale, profit, revenue) that a business wishes to monitor.
+
+**Data Modeling** - A process used to define and analyze data requirements
+needed to support the business processes within the scope of
+corresponding information systems in organizations.
+
+**Data Modeling Standards** - The most basic and standard design principles
+for data modeling. These standards must be adhered to when creating new
+models or updating existing models in the Open edX ecosystem.
+
+**IDA** - Independently Deployed Application. Similar to a microservice or
+stand alone application.
+
+
+Open edX Ecosystem
+------------------
+
+Used in this document, the phrase "Open edX Ecosystem" includes the
+services that comprise and support the running of an Open edX
+installation (e.g.,. edx.org). This includes:
+
+- The LMS and Studio
+
+- IDAs such as Notes, Ecommerce, and Forums
+
+- Backend systems such as data analytics and operational logging
+
+- Third party services used to extend functionality of applications, such as Optimizely, Google Analytics, and Sailthru
+
+
+Specification
+=============
+
+Guiding Principles
+------------------
+This section is intended to give the reader a framework to think about data modeling at edX.
+
+Data is an asset
+~~~~~~~~~~~~~~~~
+
+edX uses data to decide which people should receive marketing emails,
+who passes or fails a course, or how much to pay our partners. Our
+partners use our data to target users and to
+refine and improve their courses. The data we collect today is being
+used to advance academic research about online learning and pedagogy.
+Decision making at edX should be data-driven and based on this collected data.
+
+This data is one of our most valuable assets and it
+should be a first-order concern. Save everything (disk is cheap). The more data we are able to collect about users,
```

Comment:
Citation needed.  What is cheap? Edx.org has multiple terrabytes of history data that is running on expensive "hot" databases, never archived, snapshotted daily, and that adds a significant management overhead to every database maintenance operation. This costs us tens of thousands of dollars per month.  Do we ever look at the data? Anecdotally I can say we don't.